osf/models/registrations.py

```diff
@@ -432,28 +440,71 @@ class DraftRegistrationLog(ObjectIDMixin, BaseModel):
     draft = models.ForeignKey('DraftRegistration', related_name='logs',
                               null=True, blank=True, on_delete=models.CASCADE)
-    user = models.ForeignKey('OSFUser', null=True, on_delete=models.CASCADE)
+    user = models.ForeignKey('OSFUser', db_index=True, null=True, blank=True, on_delete=models.CASCADE)
+    params = DateTimeAwareJSONField(default=dict)
 
     SUBMITTED = 'submitted'
     REGISTERED = 'registered'
     APPROVED = 'approved'
     REJECTED = 'rejected'
 
+    EDITED_TITLE = 'edit_title'
+    EDITED_DESCRIPTION = 'edit_description'
+    CATEGORY_UPDATED = 'category_updated'
+
+    CONTRIB_ADDED = 'contributor_added'
+    CONTRIB_REMOVED = 'contributor_removed'
+    CONTRIB_REORDERED = 'contributors_reordered'
+    PERMISSIONS_UPDATED = 'permissions_updated'
+
+    MADE_CONTRIBUTOR_VISIBLE = 'made_contributor_visible'
+    MADE_CONTRIBUTOR_INVISIBLE = 'made_contributor_invisible'
+
+    AFFILIATED_INSTITUTION_ADDED = 'affiliated_institution_added'
+    AFFILIATED_INSTITUTION_REMOVED = 'affiliated_institution_removed'
+
+    CHANGED_LICENSE = 'license_changed'
+
+    TAG_ADDED = 'tag_added'
+    TAG_REMOVED = 'tag_removed'
+
     def __repr__(self):
         return ('<DraftRegistrationLog({self.action!r}, date={self.date!r}), '
                 'user={self.user!r} '
                 'with id {self._id!r}>').format(self=self)
 
+    class Meta:
+        ordering = ['-created']
+        get_latest_by = 'created'
+
+
+class DraftRegistration(ObjectIDMixin, DirtyFieldsMixin, BaseModel, Loggable,
+        EditableFieldsMixin, GuardianMixin):
+    # Fields that are writable by DraftRegistration.update
+    WRITABLE_WHITELIST = [
+        'title',
+        'description',
+        'category',
+        'node_license',
+    ]
 
-class DraftRegistration(ObjectIDMixin, BaseModel):
     URL_TEMPLATE = settings.DOMAIN + 'project/{node_id}/drafts/{draft_id}'
+    # Overrides EditableFieldsMixin to make title not required
+    title = models.TextField(validators=[validate_title], blank=True, default='')
+
+    _contributors = models.ManyToManyField(OSFUser,
+                                           through=DraftRegistrationContributor,
+                                           related_name='draft_registrations')
+    affiliated_institutions = models.ManyToManyField('Institution', related_name='draft_registrations')
+    node_license = models.ForeignKey('NodeLicenseRecord', related_name='draft_registrations',
+                                     on_delete=models.SET_NULL, null=True, blank=True)
 
     datetime_initiated = NonNaiveDateTimeField(auto_now_add=True)
     datetime_updated = NonNaiveDateTimeField(auto_now=True)
     deleted = NonNaiveDateTimeField(null=True, blank=True)
 
     # Original Node a draft registration is associated with
-    branched_from = models.ForeignKey('Node', related_name='registered_draft',
+    branched_from = models.ForeignKey('AbstractNode', related_name='registered_draft',
```

Comment:
Made this more general so branched from could be a `DraftNode`.  Have considered making `DraftNode` a subclass of a `Node` as well.